vault/login_mfa.go

```diff
@@ -2653,6 +2653,31 @@ func (b *LoginMFABackend) MemDBMFALoginEnforcementConfigByNameAndNamespace(name,
 	return eConfig.Clone()
 }
 
+func (b *LoginMFABackend) MemDBMFALoginEnforcementConfigByID(id string) (*mfa.MFAEnforcementConfig, error) {
+	if id == "" {
+		return nil, fmt.Errorf("missing config id")
+	}
+
+	txn := b.db.Txn(false)
+	defer txn.Abort()
+
+	eConfigRaw, err := txn.First(memDBMFALoginEnforcementsTable, "id", id)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch MFA login enforcement config from memdb using id: %w", err)
+	}
+
+	if eConfigRaw == nil {
+		return nil, nil
```

Comment:
> Function MemDBMFALoginEnforcementConfigByID can return an error, and has a statement that returns only nils

Should we address this here and return an error instead of nil? Something like "fmt.Errorf("could not find config wth id")"

Reply:
This matches the existing implementation. I agree this isn't great and should be addressed but this would require updates to not only this method, but the other methods as well as the callers in order to better handle suddenly returning an error here as the callers handle this case specifically which I think is a much larger change and outside of the intended scope of the fix these methods facilitate.